src/main/java/org/cardanofoundation/hydra/client/HydraWSClient.java

```diff
@@ -61,6 +64,11 @@ public void onMessage(String message) {
 
         val raw = MoreJson.read(message);
         val tagString = raw.get("tag").asText();
+        val seq = raw.get("seq").asInt();
+        if (seq < fromSeq) {
```

Comment:
Is it possible to get correct Hydra state if we return here ? Or, should we just skip listener invocation when seq < fromSeq ?

Reply:
good comment.